jina/clients/base/http.py

```diff
@@ -1,26 +1,68 @@
-import asyncio
-from contextlib import nullcontext
-from typing import Callable, Optional
+from typing import Callable, Dict, Optional
+from contextlib import nullcontext, AsyncExitStack
 
 from ..base import BaseClient, InputType
 from ..helper import callback_exec
 from ...excepts import BadClient
 from ...importer import ImportExtensions
 from ...logging.profile import ProgressBar
 from ...types.request import Request
+from ...peapods.runtimes.prefetch.client import HTTPClientPrefetcher
+
+
+class HTTPClientlet:
+    """HTTP Client to be used with prefetcher"""
+
+    def __init__(self, url: str) -> None:
+        """HTTP Client to be used with prefetcher
+
+        :param url: url to send POST request to
+        """
+        self.url = url
+        self.msg_recv = 0
+        self.msg_sent = 0
+        self.session = None
+
+    async def send_message(self, request: Dict):
+        """Sends a POST request to the server
+
+        :param request: request as dict
+        :return: send post message
+        """
+        return await self.session.post(url=self.url, json=request).__aenter__()
+
+    async def __aenter__(self):
+        """enter async context
+
+        :return: start self
+        """
+        return await self.start()
+
+    async def start(self):
+        """Create ClientSession and enter context
+
+        :return: self
+        """
+        import aiohttp
+
+        self.session = aiohttp.ClientSession()
+        await self.session.__aenter__()
```

Comment:
to handle this, use the `ExitStack` inheritance pattern? Maybe there is some AsyncExitStack?

Reply:
I faced issues with `aiohttp.ClientSession()` while reusing the `AsyncExitStack` defined [here](https://github.com/jina-ai/jina/blob/feat-http-client-prefetch/jina/clients/base/http.py#L88), since that is the caller for `HTTPClientlet`. I'll recheck this while refactoring on `WebsocketBaseClient`, as that'd be adapted `aiohttp` in https://github.com/jina-ai/jina/issues/3563